apps/mobile/src/screens/chat/chat.tsx

```diff
@@ -1,72 +1,96 @@
-import {
-	CheckboxCategoriesForm,
-	ScreenWrapper,
-	ScrollView,
-	Text,
-	View,
-} from "~/libs/components/components";
-import { BaseColor } from "~/libs/enums/enums";
+import { FlatList, ScreenWrapper, View } from "~/libs/components/components";
 import {
 	useAppDispatch,
 	useAppSelector,
 	useCallback,
 	useEffect,
-	useState,
 } from "~/libs/hooks/hooks";
 import { globalStyles } from "~/libs/styles/styles";
-import { type CategoriesSelectedRequestDto } from "~/libs/types/types";
-import { actions as categoriesActions } from "~/slices/quiz/quiz";
+import {
+	type ChatMessageDto,
+	type TaskMessage,
+	type TextMessage,
+} from "~/packages/chat/chat";
+import { actions as chatActions } from "~/slices/chat/chat";
 
-import { ChatBox } from "./libs/components/components";
+import {
+	ChatBox,
+	ChatButtons,
+	InitialChatMessage,
+	Loader,
+} from "./libs/components/components";
 import { styles } from "./styles";
 
+type ChatMessageItem = Omit<
+	ChatMessageDto<TaskMessage[] | TextMessage>,
+	"createdAt" | "id"
+>;
+
 const Chat: React.FC = () => {
-	const [submittedCategoryIds, setSubmittedCategoryIds] = useState<number[]>(
-		[],
-	);
 	const dispatch = useAppDispatch();
-	const categories = useAppSelector((state) => state.quiz.scores);
+
+	const threadId = useAppSelector((state) => state.chat.threadId);
+	const dataStatus = useAppSelector((state) => state.chat.dataStatus);
+	const messages = useAppSelector((state) => state.chat.messages);
 
 	useEffect(() => {
-		void dispatch(categoriesActions.getScores());
+		void dispatch(chatActions.initConversation());
 	}, [dispatch]);
 
-	const handleRetakeQuizSubmit = useCallback(
-		(payload: CategoriesSelectedRequestDto): void => {
-			setSubmittedCategoryIds(payload.categoryIds);
-			// TODO: send selectedCategoriesSubmissionData to backend
+	const isLoading = dataStatus === "pending";
+
+	const handleRenderItem = useCallback(
+		({ item }: { item: ChatMessageItem }) => {
+			return (
+				<ChatBox author={item.author} payload={item.payload} type={item.type} />
+			);
 		},
 		[],
 	);
 
+	const handleKeyExtractor = useCallback(
+		(item: ChatMessageItem, index: number) => index.toString(),
+		[],
+	);
+
 	return (
-		<ScreenWrapper style={[globalStyles.flex1, styles.container]}>
-			<ScrollView
+		<ScreenWrapper edges={[]} style={[globalStyles.flex1, styles.container]}>
```

Comment:
```suggestion
		<ScreenWrapper style={[globalStyles.flex1, styles.container]}>
```
Can we remove it, if we don't pass anything?

Reply:
Not resolved